app/controllers/admin/routes.py

```diff
@@ -74,9 +72,15 @@ def createEvent(templateid, programid=None):
     if request.method == "POST":
         eventData.update(request.form.copy())
 
-    if program:
-        # TODO need to handle the multiple programs case
+    if program and request.method == "GET":
         eventData["program"] = program
+        eventData['contactName'] = "CELTS Admin"
+        eventData['contactEmail'] = app.config['celts_admin_contact']
```

Comment:
They are still dependent on a program existing. You'll have to split up your conditional so we can handle get request and program vs get request and no program